### PR TITLE
azure-cosmos-spark release 2024-03-01

### PR DESCRIPTION
**Pull request by FabianMeiswinkel**
Fixed the UserAgent suffix when using a client retrieved via UDF `CosmosAsyncClientCache.getCosmosClientFuncFromCache` on an executor.

### CODE REVIEW

File: sdk/cosmos/azure-cosmos-spark_3_2-12/src/main/scala/com/azure/cosmos/implementation/SparkBridgeImplementationInternal.scala [hunk 1]
```diff
@@ -361,14 +361,14 @@ private[cosmos] object SparkBridgeImplementationInternal extends BasicLoggingTra
   }
 
   def overrideDefaultTcpOptionsForSparkUsage(): Unit = {
-    val overrideJson = "{\"timeoutDetectionEnabled\": true, \"timeoutDetectionDisableCPUThreshold\": 70.0," +
-      "\"timeoutDetectionTimeLimit\": \"PT600S\", \"timeoutDetectionHighFrequencyThreshold\": 100," +
+    val overrideJson = "{\"timeoutDetectionEnabled\": true, \"timeoutDetectionDisableCPUThreshold\": 75.0," +
```
Comment (xinlian12) [inline sdk/cosmos/azure-cosmos-spark_3_2-12/src/main/scala/com/azure/cosmos/implementation/SparkBridgeImplementationInternal.scala:364]: are these change intentional?

File: sdk/cosmos/azure-cosmos-spark_3_2-12/src/main/scala/com/azure/cosmos/spark/udf/CosmosAsyncClientCache.scala [hunk 2]
```diff
@@ -20,13 +20,13 @@ object CosmosAsyncClientCache {
...
-      effectiveUserConfig,
-      useEventualConsistency = false,
-      CosmosClientConfiguration.getSparkEnvironmentInfo(SparkSession.getActiveSession))
+    val readConfig = CosmosReadConfig.parseCosmosReadConfig(effectiveUserConfig)
+    val useEventualConsistency = readConfig.forceEventualConsistency
+    val sparkEnvironmentInfo = CosmosClientConfiguration.getSparkEnvironmentInfo(SparkSession.getActiveSession)
 
-    new CosmosAsyncClientCacheItem(
-      CosmosClientCache(cosmosClientConfig, None, "CosmosAsyncClientCache.getCosmosClientFromCache"))
+    // delay getting the client from cache here to allow this to be executed on executors
```
Comment (xinlian12) [inline sdk/cosmos/azure-cosmos-spark_3_2-12/src/main/scala/com/azure/cosmos/spark/udf/CosmosAsyncClientCache.scala:27]: nit: remove the comment as in this method the client creation will not be delayed?

File: sdk/cosmos/azure-cosmos-spark_3_2-12/src/main/scala/com/azure/cosmos/implementation/SparkBridgeImplementationInternal.scala (hunk 1 above)
Comment (FabianMeiswinkel) [reply sdk/cosmos/azure-cosmos-spark_3_2-12/src/main/scala/com/azure/cosmos/implementation/SparkBridgeImplementationInternal.scala:364]: yes - want your eyes on them especially. Background - there is sometimes a straggling single worker - form the errors because it has transit timeouts to single partition - backend is fine (other executors succeeed) - so these changes try to make a better compromise between not being too noisy with bad connection detection but still be able to at least identify bad connection in 1-2 minutes

File: sdk/cosmos/azure-cosmos-spark_3_2-12/src/main/scala/com/azure/cosmos/spark/udf/CosmosAsyncClientCache.scala (hunk 2 above)
Comment (FabianMeiswinkel) [reply sdk/cosmos/azure-cosmos-spark_3_2-12/src/main/scala/com/azure/cosmos/spark/udf/CosmosAsyncClientCache.scala:27]: done

**Review by xinlian12 — APPROVED** [review]
LGTM, thanks
